File: xkcd/src/main/java/xyz/jienan/xkcd/home/base/ContentMainBasePresenter.java
package xyz.jienan.xkcd.home.base;

import xyz.jienan.xkcd.base.BasePresenter;

public interface ContentMainBasePresenter extends BasePresenter {

    void favorited(long currentIndex, boolean isFav);

    void liked(long currentIndex);

    void setLastViewed(int lastViewed);

    void getInfoAndShowFab(int currentIndex);

    int getLatest();

    void setLatest(int latestIndex);

    int getLastViewed(int latestIndex);

    void loadLatest();

    void searchContent(String query);

    long getRandomUntouchedIndex();
}

File: xkcd/src/main/java/xyz/jienan/xkcd/home/base/ContentMainBasePresenter.kt
package xyz.jienan.xkcd.home.base

import xyz.jienan.xkcd.base.BasePresenter

interface ContentMainBasePresenter : BasePresenter {
    fun favorited(currentIndex: Long, isFav: Boolean)

    fun liked(currentIndex: Long)

    fun setLastViewed(lastViewed: Int)

    fun getInfoAndShowFab(currentIndex: Int)

    var latest: Int

    fun getLastViewed(latestIndex: Int): Int

    fun loadLatest()

    fun searchContent(query: String)

    val randomUntouchedIndex: Long
}